### PR TITLE
Support new field operators

### PR DESCRIPTION
**Pull request by dimagorelik**
Support new field operators:
1. IsNotNull - generate <> SQL condition
2. eq between two fields - generate f1.eq(f2)

### CODE REVIEW

File: main/src/main/java/com/kenshoo/pl/entity/EntityField.java [hunk 1]
```diff
@@ -36,6 +36,18 @@ default PLCondition eq(T value) {
...
+    default PLCondition eq(EntityField<?, T> otherField) {
+        if (isVirtual()) {
+            throw new UnsupportedOperationException("The equals operation is unsupported for virtual fields");
+        }
+
+        @SuppressWarnings("unchecked")
+        final TableField<Record, T> tableField = (TableField<Record, T>)getDbAdapter().getFirstTableField();
+        @SuppressWarnings("unchecked")
+        final TableField<Record, T> otherTableField = (TableField<Record, T>)otherField.getDbAdapter().getFirstTableField();
+        return new PLCondition(tableField.eq(otherTableField), entity -> entity.safeGet(this).equalsValue(entity.safeGet(otherField).get()), this, otherField);
```
Comment (effiban) [inline main/src/main/java/com/kenshoo/pl/entity/EntityField.java:48]: Please use `equals()` instead of `equalsValue()` - you are comparing two Triptionals here,  so it needs to be a regular comparison.

File: integration-tests/src/test/java/com/kenshoo/pl/entity/PLContextSelectTest.java [hunk 2]
```diff
@@ -96,12 +96,38 @@ public void selectFromSingleEntity() {
...
+                .where(TestParentEntityType.FIELD1.eq(TestParentEntityType.FIELD2))
+                .fetch();
+        assertThat("Incorrect number of entities fetched: ",
+                entities.size(), is(1));
+
+        assertThat(entities.get(0),
+                hasFieldValues(fieldValue(TestParentEntityType.ID, 2),
+                        fieldValue(TestParentEntityType.FIELD1, "ParentBravo"),
+                        fieldValue(TestParentEntityType.FIELD2, "ParentBravo")));
+    }
```
Comment (effiban) [inline integration-tests/src/test/java/com/kenshoo/pl/entity/PLContextSelectTest.java:112]: Please add a test for the case where both are `NULL`

File: integration-tests/src/test/java/com/kenshoo/pl/entity/PLContextSelectTest.java [hunk 3]
```diff
@@ -96,12 +96,38 @@ public void selectFromSingleEntity() {
...
+
+    @Test
+    public void selectIsNotNull() {
+        final List<CurrentEntityState> entities = plContext.select(TestParentEntityType.FIELD1)
+                .from(TestParentEntityType.INSTANCE)
+                .where(TestParentEntityType.FIELD1.isNotNull())
+                .fetch();
         assertThat("Incorrect number of entities fetched: ",
                 entities.size(), is(2));
```
Comment (effiban) [inline integration-tests/src/test/java/com/kenshoo/pl/entity/PLContextSelectTest.java:132]: Please add a test for the opposite case where the field is `NULL`

File: main/src/test/java/com/kenshoo/pl/entity/EntityFieldConditionsTest.java [hunk 4]
```diff
@@ -39,6 +39,27 @@ public void testFieldInValuesInPostFetchCondition() {
...
+
+        assertThat(TestEntityType.NAME1.eq(TestEntityType.NAME2).getPostFetchCondition().test(entity), is(true));
+    }
+
+    @Test
+    public void testTwoFieldNotEqualInPostFetchCondition() {
+        final var entity = createEntityWith(TestEntityType.NAME1, "myName", TestEntityType.NAME2, "anotherName");
+
+        assertThat(TestEntityType.NAME1.eq(TestEntityType.NAME2).getPostFetchCondition().test(entity), is(false));
+    }
```
Comment (effiban) [inline main/src/test/java/com/kenshoo/pl/entity/EntityFieldConditionsTest.java:61]: please add more tests for checking equals when:
- exactly one is NULL
- both are NULL

File: main/src/main/java/com/kenshoo/pl/entity/EntityField.java (hunk 1 above)
Comment (dimagorelik) [reply main/src/main/java/com/kenshoo/pl/entity/EntityField.java:48]: @effiban 
Do you mean 
entity.safeGet(this).equals(entity.safeGet(otherField).get())?

File: integration-tests/src/test/java/com/kenshoo/pl/entity/PLContextSelectTest.java (hunk 2 above)
Comment (dimagorelik) [reply integration-tests/src/test/java/com/kenshoo/pl/entity/PLContextSelectTest.java:112]: @effiban 
This case is not supported by MySql

Comment (effiban) [reply integration-tests/src/test/java/com/kenshoo/pl/entity/PLContextSelectTest.java:112]: @dimagorelik - sorry, you're right. I was mixing up the two cases, when we translate to SQL and when we do a comparison in Java only.

**Review by effiban — APPROVED** [review]

**Review by effiban — APPROVED** [review]
